client/src/components/App.jsx

```diff
@@ -8,14 +8,68 @@ export default class App extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      product_id: new URLSearchParams(window.location.search).get('id')
+      product_id: new URLSearchParams(window.location.search).get('id'),
+      cart: {},
+      outfit: []
+    }
+    this.addToCart = this.addToCart.bind(this);
+    this.removeFromCart = this.removeFromCart.bind(this);
+    this.addToOutfit = this.addToOutfit.bind(this);
+    this.removeFromOutfit = this.removeFromOutfit.bind(this);
+  }
+
+  addToCart(sku, qty) {
+    let currentQuantityInCart = this.state.cart[sku] || 0;
+    this.setState({
+      cart: {
+        ...this.state.cart,
+        [sku]: currentQuantityInCart + qty
+      }
+    });
+  }
+
+  // TODO - delete item if none left in cart...currently the cart will hold
+  // the sku with a quantity of 0
+  removeFromCart(sku, qty) {
+    if (!this.state.cart[sku]) return;
+    let newQuantity = this.state.cart[sku] - qty;
+    newQuantity = newQuantity > 0 ? newQuantity : 0;
+    this.setState({
```

Comment:
Do you plan on just using pop to remove the quantity with 0? You might be able to fit it in a single line.